source/history.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+var noop = function () {
+    return false;
+};
+var pushState = window.history.pushState;
+
+/**
+ * History
+ * @type {Object}
+ */
+var history = {
+    /**
+     * Called everytime the page url's been changed.
+     * Does nothing by default. Consumer has to define the behaviour.
+     * @type {Function}
+     */
+    onChangesUrl: noop,
+};
+
+window.history.pushState = function () {
+    // the browser should change the url first
+    setTimeout(history.onChangesUrl, 0);
+    return pushState.apply(window.history, arguments);
+};
+window.addEventListener('popstate', function () {
```

Comment:
🐔 

Performing side effects when requiring the module seems weird to me. These side effects are implicit, and I can’t tell that `require('./history')` does any side effects by just looking on it. I’d better move all the initialization code into `history.init()` or something similar to make the initialization explicit. 

What do you think?